<!--
  Copyright 2016 The Cartographer Authors

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->

<launch>
  <param name="robot_description"
    textfile="$(find cartographer_ros)/urdf/cinderella.urdf" />

  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />

  <!-- publish odom with twist in body frame to base_link -->
  <node name="odom_in_body" pkg="odom_relay"
    type="odom_relay_twist_body" />

  <!-- Publish tf from odometry origin to imu link -->
  <node pkg="odom_relay" type="odom_relay_tf" name="odom_tf"
    output="screen">
    <param name="odom_topic" value="/vislam/odometry"/> 
    <param name="frame_id" value="vislam_odom"/> 
    <param name="child_frame_id" value="odom_frame"/> 
  </node>

  <node name="cartographer_node" pkg="cartographer_ros"
      type="cartographer_node" args="
          -configuration_directory $(find cartographer_ros)/configuration_files
          -configuration_basename my_robot.lua"
      output="screen">
    <!-- <remap from="points2_1" to="laser/cloud" /> -->
    <remap from="scan" to="/laser/scan_laser_link" />
    <remap from="imu" to="/mavros/imu/data" />
    <remap from="odom" to="/vislam/odometry_body" />

    <param name="/map_at_launch" value="false"/>

    <!-- Time-constant for the drift estimation lowpass filter -->
    <param name="/lowpass_time_constant" value="0.5"/>
  </node>

  <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
      type="cartographer_occupancy_grid_node" args="-resolution 0.05" />
</launch>
